subsys/bluetooth/mesh/transport.c

```diff
@@ -1454,6 +1477,7 @@ static int trans_seg(struct net_buf_simple *buf, struct bt_mesh_net_rx *net_rx,
 
 	if (rpl) {
 		bt_mesh_rpl_update(rpl, net_rx);
+		rpl->seg = auth_seqnum;
```

Comment:
If source node send single segmented-message and also is first communication with this node address, then `if (rpl) `would be **false**, and duplicate check will be invalid?

Reply:
I don't believe so. `bt_mesh_rpl_check` returns a valid `rpl` pointer even when it's the first message. It's only if the check fails or the message is received on behalf of a friend that the pointer isn't set.